### DISCUSSION TITLE
Trying to start min.io on win2019

### DISCUSSION THREAD

**Discussion by codeninja75** [minio/minio#13621] [Q&A] [answered]
So I'm trying to start a min.io standalone box.  Got a win 2019 server , I can run it fine and it presents the proper cert on the console.  The cert is a wildcard, and when I login it bombs w/ Post "https://10.10.1.30:9000/": x509: cannot validate certificate for 10.10.1.30 because it doesn't contain any IP SANs.  So I'm trying to launch it with a cmd line to tell it what 'name' it should be using.  min.io server https://minio-is-cool.bob.com/ D:\data --console-address ":9001" but that errors with incorrect number of endpoints provided.  Is this a possible config?  Trying to test out secure deployment in our dev env before we set it up in prod.

**Comment by harshavardhana** [comment, marked as answer]
> So I'm trying to start a min.io standalone box. Got a win 2019 server , I can run it fine and it presents the proper cert on the console. The cert is a wildcard, and when I login it bombs w/ Post "https://10.10.1.30:9000/": x509: cannot validate certificate for 10.10.1.30 because it doesn't contain any IP SANs. So I'm trying to launch it with a cmd line to tell it what 'name' it should be using. min.io server https://minio-is-cool.bob.com/ D:\data --console-address ":9001" but that errors with incorrect number of endpoints provided. Is this a possible config? Trying to test out secure deployment in our dev env before we set it up in prod.

You need to make sure to set what is your domain for TLS certs  set correct

- MINIO_SERVER_URL
- MINIO_BROWSER_REDIRECT_URL

Read fully https://github.com/minio/minio#things-to-consider

**Comment by codeninja75** [comment]
That was very helpful.  It seems to be able to understand the wildcard cert now.  Now it's just giving me an connection refused on 443 

Post "https://cool.bob.com/": dial tcp [fe80::ec31:765b:9394:bfa% (VLAN 136)]:443: connectex: No connection could be made because the target machine actively refused it.   When I run Get-NetTCPConnection i see the process listening on 9001 (where the console does come up) and on port 9000.  So I had to set the MINIO_BROWSER_REDIRECT_URL to https://cool.bob.com and set the MINIO_SERVER_URL to https://cool.bob.com:9000 and looks as if I'm in.  Thanks for the help.  Is there a canonical listing of all the environment variables that min.io uses the server?

**Reply by harshavardhana** [reply]
> Post "https://cool.bob.com/": dial tcp [fe80::ec31:765b:9394:bfa% (VLAN 136)]:443: connectex: No connection could be made because the target machine actively refused it. When I run Get-NetTCPConnection i see the process listening on 9001 (where the console does come up) and on port 9000. So I had to set the MINIO_BROWSER_REDIRECT_URL to https://cool.bob.com and set the MINIO_SERVER_URL to https://cool.bob.com:9000 and looks as if I'm in. Thanks for the help. Is there a canonical listing of all the environment variables that min.io uses the server?

Of course your MinIO is not running on port 443
